common/paramsbuilder/module.go

```diff
@@ -11,52 +13,57 @@ var ErrNoSupportedModule = errors.New("no supported module was chosen")
 // This is relevant where there are several APIs for different sub-products, and the APIs
 // are versioned differently or have different ways of constructing URLs and requests for reading/writing.
 type Module struct {
-	// Name represents the name of the sub-product. e.g. crm, marketing, etc.
-	Name string
+	// ID is name of the module.
```

Comment:
1. There's different schools of thoughts, but `ID` conveys that this is an acronym which is not the case, so we tend to use `Id` instead

2. `Id is a unique identifier for the module. It makes it easier to group multiple endpoints under the same module, even if they do not share a common URL prefix.`

Reply:
ID is abbreviation for identifier.
I think creating a difference between ID and Id creates confusion. Lauren pointed out that it should be `Id` for JSON fields. In code I commit this sin. ID visually draws attention and looks like a concept. In the Java world it was always Id, in go world I see ID. I think my mental mapping got messed up since then.

It is a go convention to do it that way. Linters complain about such things (of course not in comments ahaha).

https://google.github.io/styleguide/go/decisions#initialisms, id or ID.

Reply:
Also in the IDE it highlights field names in comments if the spelling matches. Thats why I wrote ID as caps.

![image](https://github.com/user-attachments/assets/53feb3ae-5740-4611-9f55-053c847312c2)